docs/UIKit/ConversationListView.md

```diff
@@ -0,0 +1,82 @@
+## Conversation List View
+==========================
+
+Conversation List View display a list of conversations in which the user is
+participanting.
+
+You can customize the Conversation List View by creating your own view
+controller that extends the Conversation List View. Make sure you set your own
+view controller as the `Custom Class` in your storyboard.
+
+```swift
+import SKYKitChat
+
+class ConversationListDemoViewController: SKYChatConversationListViewController {
+
+}
+```
+
+To customize the Conversation List View, you can override some of its methods:
+
+- `performQuery()`
+- `performUserQuery(byIDs:)`
+- `handleQueryResult(result:)`
+- `handleQueryError(error:)`
+- `handleUserQueryResult(result:)`
+- `handleUserQueryError(error:)`
+
+Besides overriding these methods, you can also implement some methods
+of SKYChatConversationListViewControllerDataSource to
+provide extra data for displaying a conversation.
+
+For example, You may provide custom avatar of the conversation as followed:
+
+```swift
+class ConversationListDemoViewController: SKYChatConversationListViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.dataSource = self
+    }
+}
+
+extension ConversationListDemoViewController: SKYChatConversationListViewControllerDataSource {
+    func listViewController(_ controller: SKYChatConversationListViewController,
+                            avatarImageForConversation conversation: SKYConversation,
+                            atIndexPath indexPath: IndexPath) -> UIImage?
+    {
+        // implement a store for storing conversation data
+        var image = AvatarStore.shared.avatar(forConversation: conversation)
+        if image == nil {
+            image = /* fetch the image */
```

Comment:
this example is not very good when it comes to fetching and updating async image, plus i prefer example to compile cleanly, which this snippet is clearly not.